frontend/src/components/GroupFilterCard.tsx

```diff
@@ -0,0 +1,172 @@
+      </Box>
+      <GroupFilterCardCompartmentValues
+        groupFilter={props.groupFilter}
+        groupFilterIndex={props.groupFilterIndex}
+        scenarioId={props.scenarioId}
+      />
+    </Box>
+  );
+}
+
+/**
+ * This component renders all compartment values of a group filter in one column.
+ */
+function GroupFilterCardCompartmentValues(props: GroupFilterCardProps): JSX.Element | null {
+  const {t, i18n} = useTranslation();
+  const theme = useTheme();
+
+  const {formatNumber} = NumberFormatter(i18n.language, 1, 0);
+
+  const groupFilterList = useAppSelector((state) => state.dataSelection.groupFilters);
+  const day = useAppSelector((state) => state.dataSelection.date);
+  const node = useAppSelector((state) => state.dataSelection.district?.ags);
+  const selectedCompartment = useAppSelector((state) => state.dataSelection.compartment);
+  const compartmentsExpanded = useAppSelector((state) => state.dataSelection.compartmentsExpanded);
+  const compartments = useAppSelector((state) => state.scenarioList.compartments);
+
+  const {data: groupFilterData} = useGetSingleGroupFilterDataQuery(
+    {
+      id: props.scenarioId,
+      node: node,
+      day: day ?? '',
+      groupFilter: props.groupFilter,
+    },
+    {skip: !day}
+  );
+
+  const groupCompartmentsRef = useRef<Array<HTMLUListElement | null>>([]);
+
+  useEffect(() => {
+    if (groupFilterList) {
+      groupCompartmentsRef.current.slice(0, Object.values(groupFilterList).length);
+    }
+  }, [groupFilterList]);
+
+  const getGroupValue = (compartment: string): string => {
+    if (!groupFilterData) {
+      return t('no-data');
+    }
+
+    const groupFilterResults = groupFilterData.results;
+    if (groupFilterResults.length === 0 || !(compartment in groupFilterResults[0].compartments)) {
+      return t('no-data');
+    }
+
+    return formatNumber(groupFilterResults[0].compartments[compartment]);
+  };
+
+  if (groupFilterData) {
+    return (
+      <ScrollSyncPane group='compartments'>
+        <List
+          id={`scenario-card-${props.scenarioId}-group-filter-compartment-list-${props.groupFilterIndex}`}
+          className='hide-scrollbar'
+          ref={(el) => (groupCompartmentsRef.current[props.groupFilterIndex] = el)}
+          dense={true}
+          disablePadding={true}
+          sx={{
+            maxHeight: compartmentsExpanded ? '248px' : 'auto',
+            overflowX: 'hidden',
+            overflowY: 'auto',
+          }}
+        >
+          {compartments.map((compartment, i) => {
+            return (
+              // hide compartment if compartmentsExpanded false and index > 4
+              // highlight compartment if selectedCompartment === compartment
+              <ListItem
+                key={compartment}
+                sx={{
+                  display: compartmentsExpanded || i < 4 ? 'flex' : 'none',
+                  color: selectedCompartment === compartment ? theme.palette.text.primary : theme.palette.text.disabled,
```

Comment:
Move comments to their respective code.
```suggestion
              <ListItem
                key={compartment}
                sx={{
                  // hide compartment if compartmentsExpanded false and index > 4
                  display: compartmentsExpanded || i < 4 ? 'flex' : 'none',
                  // highlight compartment if selectedCompartment === compartment
                  color: selectedCompartment === compartment ? theme.palette.text.primary : theme.palette.text.disabled,
```